Docs/ArchitectureDecisions/003_documentation.md

```diff
@@ -0,0 +1,42 @@
+# 003 Documentation
+
+**Feature**: [#700](https://github.com/Azure/iotedge-lorawan-starterkit/issues/700)  
+**Authors**: Roel Fauconnier  
+**Status**: Proposes
+
+## Overview / Problem Statement
+
+Currently the documentation is not very well structured, and content is hard to discover. There is no search except for the built-in GitHub code search. Some of the documentation is spread out across different files.
+
+## Possible solutions
+
+### 1. Structurize the current folders
+
+Bring more structure to the current documentation folders, and use GitHub "markdown-preview" feature. The docs can be searched via GitHub search function. This also means adding additional links from each document to other documents to make things more discoverable.
+
+### 2. Use GitHub pages
+
+GitHub has a "Pages" functionality which allows you to publish a static website to a predefined URL: `https://projectname.github.io`. It can be published from a branch and/or folder. The folders are limited to `/` or `/docs`.  
+The best practice seems to be to create a detached branch or git subtree called `gh-pages` which contains only the static website in a `/docs` folder.
+
+#### 2.1 Use Jekyll static site generator
```

Comment:
I tried this option a few years ago and can't say I was super convinced - it worked but Jekyll has a very small community.